evolver/calibration/standard/actions/temperature.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel, Field
+
+from evolver.calibration.action import CalibrationAction, CalibrationActionModel
+from evolver.calibration.interface import ProcedureStateModel
+
+
+class VialData(BaseModel):
+    reference: List[float] = Field(default_factory=list, description="Reference temperature values.")
+    raw: List[float] = Field(default_factory=list, description="Raw temperature readings.")
+    fit: Dict = Field(
+        default_factory=dict,
+        description="Fit parameters for the temperature sensor, calculated using the output transformer defined "
+        "for the vial and the reference and raw values collected in the procedure.",
+    )
+
+
+class ProcedureState(ProcedureStateModel):
+    """State model for the temperature calibration procedure."""
+
+    selected_vials: List[int] = Field(
+        default_factory=list, description="List of vials identified by their index being calibrated."
+    )
+    vial_data: Dict[int, VialData] = Field(
+        default_factory=dict, description="Calibration data for each vial, keyed by vial index."
+    )
+
+
+class ReferenceValueAction(CalibrationAction):
+    class FormModel(BaseModel):
+        temperature: float = Field(..., title="Temperature", description="Temperature in degrees Celsius")
+
+    def __init__(self, hardware, description: str, vial_idx: int, name: str):
+        super().__init__(CalibrationActionModel(name=name, description=description, requires_input=True))
```

Comment:
One way to handle repetition of func sig parmas in derived classes is to:
```suggestion
    def __init__(self, hardware, vial_idx: int, *args, **kwargs):
        super().__init__(CalibrationActionModel(*args, requires_input=True, **kwargs)
```
or 
```python
def __init__(self, hardware, vial_idx: int, *args, **kwargs):
        super().__init__(*args, **kwargs)
```
if the func sig for ``super`` followed a pattern to allow for arbitrary inheritance.

Reply:
Fixed this.